include/libopencm3/stm32/h7/memorymap.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBOPENCM3_MEMORYMAP_H
+#define LIBOPENCM3_MEMORYMAP_H
+
+#include <libopencm3/cm3/memorymap.h>
+
+/* --- STM32H7 specific peripheral definitions ----------------------------- */
+
+/* Memory map for all busses */
+#define PERIPH_BASE         0x40000000U
+#define PERIPH_BASE_APB1    0x40000000U
+#define PERIPH_BASE_APB2    0x40010000U
+#define PERIPH_BASE_APB3    0x50000000U
+#define PERIPH_BASE_AHB1    0x40020000U
+#define PERIPH_BASE_AHB2    0x48020000U
+#define PERIPH_BASE_AHB3    0x51000000U
+#define PERIPH_BASE_AHB4    0x58000000U
+
+/* Table 8: Register boundary addresses */
+
+/* AHB4 Peripherals */
+#define GPIO_PORT_A_BASE    0x58020000U
+#define GPIO_PORT_B_BASE    0x58020400U
+#define GPIO_PORT_C_BASE    0x58020800U
+#define GPIO_PORT_D_BASE    0x58020C00U
+#define GPIO_PORT_E_BASE    0x58021000U
+#define GPIO_PORT_F_BASE    0x58021400U
+#define GPIO_PORT_G_BASE    0x58021800U
+#define GPIO_PORT_H_BASE    0x58021C00U
+#define GPIO_PORT_I_BASE    0x58022000U
+#define GPIO_PORT_J_BASE    0x58022400U
+#define GPIO_PORT_K_BASE    0x58022800U
+#define RCC_BASE            0x58024400U
+#define POWER_CONTROL_BASE  0x58024800U
```

Comment:
I think this is just called PWR_BASE in other families isn't it? you want these names to be as common as they can be to share code

Reply:
Actually, this os only named POWER_CONTROL_BASE because it is named such on all of the other platforms.

Reply:
ok, good enough for me :)